spec/Section 2 -- Language.md

```diff
@@ -587,13 +587,13 @@ FragmentName : Name but not `on`
 
 Fragments are the primary unit of composition in GraphQL.
 
-Fragments allow for the reuse of common repeated selections of fields, reducing
-duplicated text in the document. Inline Fragments can be used directly within a
-selection to condition upon a type condition when querying against an interface
-or union.
+Each client component should declare its data needs in a dedicated fragment.
+These fragments may then be composed in the same fashion as the components
+themselves to form a GraphQL operation to issue to the server.
```

Comment:
I'm concerned that "component" here will be read too specifically (e.g. to mean a React component) rather than broadly (e.g. a "component" of client logic such as a function, class, or other...). Perhaps something along the lines of:

```suggestion
Each data-consuming component (function, class, UI element, and so on) of
a client application should declare its data needs in a dedicated fragment.
These fragments may then be composed, following the usage of the components
themselves, to form a GraphQL operation to issue to the server.
```